### PR TITLE
CIV-13201 - Add Dashboard for Defendant Response Deadline Check

### PR DESCRIPTION
**Pull request by m-meulendijks-v1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-13201

### Change description ###
When defendant does not respond in time, show this notification.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by jarekPierchala — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/DefendantResponseDeadlineScenarioTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import uk.gov.hmcts.reform.civil.sampledata.CaseDataBuilder;
+
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@Testcontainers
```
Comment (paul-pearson) [inline src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/DefendantResponseDeadlineScenarioTest.java:21]: I don't think you need these two annotations here as they are inherited from the base class.

File: src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/DefendantResponseDeadlineScenarioTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        //Verify Notification is created
+        doGet(BEARER_TOKEN, GET_NOTIFICATIONS_URL, caseId, "CLAIMANT")
+            .andExpect(status().isOk())
+            .andExpectAll(
+                status().is(HttpStatus.OK.value()),
+                jsonPath("$[0].titleEn").value("Response to the claim"),
+                jsonPath("$[0].descriptionEn").value(
+                    "<p class=\"govuk-body\">The defendant has not responded to the claim.</p>" +
+                    "<p class=\"govuk-body\">You can now request a county court judgement.<p/>" +
```
Comment (paul-pearson) [inline src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/DefendantResponseDeadlineScenarioTest.java:50]: "judgement" here should be "judgment".

File: src/main/resources/db/migration/V2024_03_25_1606__CIV-13201_response_deadline_not_responded.sql [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * Add notification template
+ */
+INSERT INTO dbs.dashboard_notifications_templates (template_name, title_En, title_Cy, description_En, description_Cy
+                                                  ,notification_role)
+VALUES ('Notice.AAA6.DefResponse.ResponseTimeElapsed.Claimant',
+        'Response to the claim', 'Response to the claim',
+        '<p class="govuk-body">The defendant has not responded to the claim.</p>'
+          '<p class="govuk-body">You can now request a county court judgement.<p/>'
```
Comment (paul-pearson) [inline src/main/resources/db/migration/V2024_03_25_1606__CIV-13201_response_deadline_not_responded.sql:17]: "judgement" here should be "judgment".

File: src/main/resources/db/migration/V2024_03_25_1606__CIV-13201_response_deadline_not_responded.sql [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+          '<p class="govuk-body">The defendant can still respond to the claim before you ask for a judgment.</p>'
+          '<p class="govuk-body"><a href="{COUNTY_COURT_JUDGEMENT_URL}" class="govuk-link">Request a CCJ</a></p>',
+        '<p class="govuk-body">The defendant has not responded to the claim.</p>'
+          '<p class="govuk-body">You can now request a county court judgement.<p/>'
```
Comment (paul-pearson) [inline src/main/resources/db/migration/V2024_03_25_1606__CIV-13201_response_deadline_not_responded.sql:21]: As above - should be "judgment".

**Review by paul-pearson — CHANGES_REQUESTED** [review]

**Review by paul-pearson — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

**Pull request merged**